plugin/core/sessions.py

```diff
@@ -71,11 +71,13 @@ def with_client(client: Client) -> 'Session':
     return session
 
 
-def get_initialize_params(project_path: str, config: ClientConfig) -> dict:
+def get_initialize_params(workspace_folders: 'List[WorkspaceFolder]', config: ClientConfig) -> dict:
```

Comment:
Instead of `List[WorkspaceFolder]`, you could use `Iterable[WorkspaceFolder]` everywhere. The first item can be retrieved with `next(iter(workspace_folders))`

Reply:
I really like this strategy for lists that are expensive-ish to compute and filtered-on multiple times when we only want 1 item.
In the case of WorkspaceFolder, we end up sending all of them anyways. 
Is there a performance benefit to programming against Iterable when receiving List? I'm a bit scared of things like empty generators being truthy, and iterators only being iterable once.